movie.sh

```diff
@@ -30,38 +27,33 @@ function main(){
     # There should be exactly one argument i.e the address path
     if [ $NUM_ARGUMENTS != 1 ]; then
         add_style "This application takes one argument : Movie Directory" 1
-        rm -rf $TEMP_PATH
         exit 4
     fi
     # Fetching Movies and removing 
-    ls $DIRECTORY_NAME > "$TEMP_PATH/movlist.txt"
-    local movie_count=$(cat "$TEMP_PATH/movlist.txt" \
+    movie_list=$(ls $DIRECTORY_NAME \
+     | grep "" | \
```

Comment:
You started your function with 4 space indentation. Every time you move another level, you add another 4 spaces. So this should be rewritten as:

```
    movie_list=$(ls $DIRECTORY_NAME \
        | grep "" \
        | sed 's/_/./g; s/ /./g')
```